rust-runtime/aws-smithy-runtime-api/src/client/auth.rs

```diff
@@ -192,6 +193,47 @@ impl From<Cow<'static, str>> for AuthSchemeId {
     }
 }
 
+impl PartialEq for AuthSchemeId {
```

Comment:
can you handle this in `::new` and normalize there? Seems simpler.

Reply:
Interesting, we need to cover the following also
```
impl From<&'static str> for AuthSchemeId {
    fn from(scheme_id: &'static str) -> Self {
        Self::new(scheme_id)
    }
}

impl From<Cow<'static, str>> for AuthSchemeId {
    fn from(scheme_id: Cow<'static, str>) -> Self {
        Self { scheme_id }
    }
}
```
but it will help eliminate handwritten trait impls and their unit tests.

Reply:
Hmm, looks like we cannot unfortunately, since `new` is a const function that cannot call a non-const operator `==`.

Reply:
Let me try something like
```
const fn normalize_auth_scheme_id(s: &'static str) -> &'static str {
    match s.as_bytes() {
        b"no_auth" => "noAuth",
        _ => s,
    }
}
```

Reply:
Thanks for the suggestion, updated in 7922ef9. Credit goes to @landonxjames for the const function above.